content/authors/client-side-validation-with-react-alert/index.md

```diff
@@ -0,0 +1,366 @@
+
+You should have an idea of the following before you read this article any further so that there wouldn’t be any form of confusion.
+
+1. We’d be needing [NodeJs](https://nodejs.org) since we’re making use of an npm package called react-alert
+
+2. An idea of how conditional statements works in react/JavaScript
+
+3. How npm works and how to install npm packages as dependencies in your projects
+
+4. A little understanding of DOM Manipulation
+
+### Overview
+
+Before React, vanillaJS was the “go-to” solution for client-side validation. One would need to understand the fundamentals of DOM manipulation and event handling.
+Since the aim of this validation process is to check if the data or information sent by the user to the server is appropriate by listening to the events that take place on the webpage.
+
+These events range from the click of a button `onClick`, the mouse hovering effect on a particular element on the page `onMouseOver`, the submission of details entered into a webform via the `onSubmit` event and many more that I can’t mention. If you want to know more about events in JavaScript, you should check them out [here](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Building_blocks/Events).
+
+Performing these tasks (DOM manipulation) in vanillaJS isn’t a big deal because the DOM is available. But in React, the DOM isn’t available a virtual DOM is available instead.
+
+Although, manipulating the DOM in React undergoes the same process of doing that in vanillaJS but there are some quirks to it. One of them is how the `document` object isn’t available in React since it is being bundled with nodejs which doesn’t run on the client-side of a web browser.
+
+Getting Started
+We’ve had a look at what we’d be needing in this article in the first section. To be able to perform client-side validation in React, in the scope of this article, let’s start by installing the necessary `npm` dependencies.
+
+I’d be making use of NextJS to bootstrap my react application because of its simplicity, its folder structure, and many other benefits of using it.
+
+You don’t need to make use of NextJS if you don’t want to, you can decide to make use of [create-react-app](https://create-react-app.dev/) or bootstrapping your react app with [ParcelJS](https://www.digitalocean.com/community/tutorials/how-to-set-up-a-react-project-with-parcel). Feel free to make use of any approach that’d be most convenient for you.
+
+To get started with NextJS, type the command below into your terminal, it will install all the dependencies that you’d need to create a React application.
+
+`npx create-next-app name-of-your-app`
+
+Since this article focuses on the use of “react-alert” for client-side validation in react. We need to also add it to the list of dependencies that’d be used in the project. The command below will handle that for us.
```

Comment:
on line 39
Use backticks to highlight keywords(react-alert npm package) i.e
```bash
`react-alert`
```

Also, still on line 39. Replace the word "it" with the "react-alert" package i.e

 We need to also add `react-alert` to the list of dependencies that’d be used in the project.